documentation/src/main/wiki/Adapter.md

```diff
@@ -28,39 +30,150 @@ is not going to provide any generic Action Adapters.
 abstraction can be used while implementing a custom Adapter but it is not required. Every Adapter must be
 exposed with a unique Event Bus address - that's the only obligation (as is the case with Knots).
 Please see an example configuration for [[Http Service Adapter|HttpServiceAdapter#how-to-configure]]
 
-## How to extend?
-Implementation of an Adapter does not require knowledge of how to communicate via the Vert.x event bus. It's wrapped by the **Vert.x Service Proxy** functionality so any new implementation can focus on the business logic of the Adapter. 
+## How to implement your own Adapter?
+Knot.x provides the [maven archetypes](https://github.com/Knotx/knotx-extension-archetype) to generate custom Adapters. 
+It is the **recommended** way to create your own Adapter.
+
+An Adapter logic is executed on a [Vert.x event loop](http://vertx.io/docs/vertx-core/java/#_reactor_and_multi_reactor). [The 
+Vert.x Golden Rule](http://vertx.io/docs/vertx-core/java/#golden_rule) says that the code should **never block** the 
+event loop. So all time consuming operations should be coded in an asynchronous way. Default Knot.x Adapters use [RxJava](http://vertx.io/docs/vertx-rx/java/) 
+which is a popular library for composing asynchronous and event-based programs using observable sequences for the Java VM.
+RxJava introduce a Reactive Programming that is a development model structured around asynchronous data streams. 
+
+| Note |
+|:------ |
+| Reactive programming code first requires a mind-shift. You are notified of asynchronous events. Then, the API can be hard to grasp (just look at the list of operators). Don’t abuse, write comments, explain, or draw diagrams. RX is powerful, abusing it or not explaining it will make your coworkers grumpy. [Read more](https://developers.redhat.com/blog/2017/06/30/5-things-to-know-about-reactive-programming/) |
 
-In order to implement an Adapter, follow the guide below:
+Implementation of an Adapter does not require knowledge of how to communicate via the Vert.x event bus.
+It's wrapped by the **Vert.x Service Proxy** functionality so any new implementation can focus on 
+the business logic of the Adapter.
 
-1. Create your Knot by extending the `io.knotx.adapter.AbstractAdapterProxy` class and implement your business logic in the `processRequest()` method with the return type of `Observable<AdapterResponse>` (promise of the `AdapterResponse`).
+In order to implement an Adapter generate a new Adapter module using maven archetype:
 
-   You can refer to `io.knotx.adapter.service.http.impl.HttpServiceAdapterProxyImpl.java` as an example.
+   `mvn archetype:generate -DarchetypeGroupId=io.knotx.archetypes -DarchetypeArtifactId=knotx-adapter-archetype -DarchetypeVersion=X.Y.Z`
 
-2. Create a class extending `AbstractVerticle` that will simply read the configuration and register your `AdapterProxy` implementation at the `address` provided.
+Note that the Adapter archetype generates both the code and all configuration
```

Comment:
Note that, the Adapter archetype generates not only a skeleton of your custom Adapter, but also all the configuration files that's required to run a Knpt.x instance.